packages/grafana-data/src/dataframe/utils.ts

```diff
@@ -26,3 +26,34 @@ export function anySeriesWithTimeField(data: DataFrame[]) {
   }
   return false;
 }
+
+/**
+ * Given data frames representing time series from different time windows, this function shifts the time values
+ * of the frames so that they are graphable.
+ * This function assumes that the first frame contains the reference time range, while consecutive frames
+ * are time shifted.
+ */
+export function shiftComparisonFramesTimestamps(frames: DataFrame[]) {
+  // First frame that holds a time field that's used as comparison diffs reference
+  let timeComparisonReferenceFrame: DataFrame | undefined;
+
+  const referenceFrame = frames[0];
+  for (let field of referenceFrame.fields) {
```

Comment:
IIUC, we only want to shift if:
1. all frames have a queryTimeRange
2. the values in the queryTimeRange are different (may not be the *same* instance)